packages/gossip/manager.go

```diff
@@ -36,16 +31,46 @@ var (
 // LoadMessageFunc defines a function that returns the message for the given id.
 type LoadMessageFunc func(messageId tangle.MessageID) ([]byte, error)
 
+// ConnectPeerOption defines an option for the DialPeer and AcceptPeer methods.
+type ConnectPeerOption func(conf *connectPeerConfig)
+
+type connectPeerConfig struct {
+	useDefaultTimeout bool
+}
+
+func buildConnectPeerConfig(opts []ConnectPeerOption) *connectPeerConfig {
+	conf := &connectPeerConfig{
+		useDefaultTimeout: true,
+	}
+	for _, o := range opts {
+		o(conf)
+	}
+	return conf
+}
+
+// WithNoDefaultTimeout returns a ConnectPeerOption that disables the default timeout for dial or accept.
+func WithNoDefaultTimeout() ConnectPeerOption {
```

Comment:
We have a default timeout inside the gossip manager that we apply by default. 
When the `WithNoDefaultTimeout` option is used, this default timeout isn't applied. 
If we rename it to `WithoutTimeout`, it might mean that there is no timeout at all like it's completely disabled, but in reality, we just don't apply the default one, and you still can set a custom timeout via context.
Other than that, it's a general naming patter for options in Go: `With...()`